scarb-metadata/src/command/metadata_command.rs

```diff
@@ -153,33 +164,69 @@ impl MetadataCommand {
 
 fn parse_stream(data: &[u8]) -> Result<Metadata, MetadataCommandError> {
     let mut err = None;
-    for line in BufRead::split(data, b'\n') {
-        let line = line?;
-
-        // HACK: Use a heuristic to guess if this line in the output is the metadata one.
-        //   This works based on the assumption that the output comes from a call to
-        //   `scarb metadata` that:
-        //   1. Used `--json` flag, so that messages are NDJSON.
-        //   2. Fields were not reordered: the "version" field is always first.
-        if !line.starts_with(br#"{"version":"#) {
-            continue;
-        }
 
-        // Deal with hypothetical case that somehow Scarb outputs other messages with "version"
-        // field first. If one is spotted before the metadata message, deserialization of it should
-        // fail. Instead of immediately returning the error, we save it, and continue to look up
-        // for the metadata later in output. If it is found -- great, we will return it and forget
-        // about the error. Otherwise, we assume that the found message was meant actually to be
-        // the metadata object that failed to deserialize, and thus we report deserialization error
-        // instead of `NotFound`.
-        match serde_json::from_slice(&line) {
-            Ok(metadata) => return Ok(metadata),
-            Err(serde_err) => err = Some(serde_err.into()),
+    let data = std::str::from_utf8(data).unwrap();
```

Comment:
Can we avoid unwrap?

Reply:
With a little bit of allocation, yes. Fixed.